src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -260,13 +268,14 @@ class ContentOpenWith extends PureComponent<Props, State> {
      * @return {void}
      */
     executeIntegrationSuccessHandler = (executeData: ExecuteAPI): void => {
+        let { integrationWindowRef }: State = this.state;
```

Comment:
I was previously maintaining a class property and the state, would that be more clear? The class property would allow us to update the window reference with a new URL/clear it when ready, and the state property tells the component when to render the portal.

Reply:
It seems like you're trying to use the existence of the window ref as a shortcut for some other state. It might clear things up to create a separate property for "show the loading indicator or not" state.